src/apprunner/wizards/apprunnerCreateServiceWizard.ts

```diff
@@ -0,0 +1,227 @@
+            Array.from(new Set(matches)).join('')
+        )
+    }
+
+    return undefined
+}
+
+function createInstanceStep(): Prompter<AppRunner.InstanceConfiguration> {
+    const enumerations = [
+        [1, 2],
+        [1, 3],
+        [2, 4],
+    ]
+
+    const items: picker.DataQuickPickItem<AppRunner.InstanceConfiguration>[] = enumerations.map(e => ({
+        label: `${e[0]} vCPU${e[0] > 1 ? 's' : ''}, ${e[1]} GBs Memory`,
+        data: { Cpu: `${e[0]} vCPU`, Memory: `${e[1]} GB` },
+    }))
+
+    return picker.createQuickPick(items, {
+        title: localize('AWS.apprunner.createService.selectInstanceConfig.title', 'Select instance configuration'),
+        buttons: makeButtons(),
+    })
+}
+
+function showDeploymentCostNotifcation(): void {
+    const shouldShow = ext.context.globalState.get('apprunner.deployments.notifyPricing', true)
+
+    if (shouldShow) {
+        const notice = localize(
+            'aws.apprunner.createService.priceNotice.message',
+            'App Runner automatic deployments incur an additional cost.'
+        )
+        const viewPricing = localize('aws.apprunner.createService.priceNotice.view', 'View Pricing')
+        const dontShow = localize('aws.generic.doNotShowAgain', "Don't Show Again")
+        const pricingUri = vscode.Uri.parse('https://aws.amazon.com/apprunner/pricing/')
+
+        vscode.window.showInformationMessage(notice, viewPricing, dontShow).then(button => {
+            if (button === viewPricing) {
+                vscode.env.openExternal(pricingUri)
+                showDeploymentCostNotifcation()
+            } else if (button === dontShow) {
+                ext.context.globalState.update('apprunner.deployments.notifyPricing', false)
+            }
+        })
+    }
+}
+
+function createSourcePrompter(
+    autoDeployButton: QuickInputToggleButton
+): Prompter<CreateServiceRequest['SourceConfiguration']> {
+    const ecrPath = {
+        label: 'ECR',
+        data: { ImageRepository: {} } as ImageRepositorySource,
+        detail: localize(
+            'AWS.apprunner.createService.ecr.detail',
+            'Create a service from a public or private Elastic Container Registry repository'
+        ),
+    }
+
+    const repositoryPath = {
+        label: 'Repository',
+        data: { CodeRepository: {} } as CodeRepositorySource,
+        detail: localize('AWS.apprunner.createService.repository.detail', 'Create a service from a GitHub repository'),
+    }
+
+    return picker.createQuickPick([ecrPath, repositoryPath], {
+        title: localize('AWS.apprunner.createService.sourceType.title', 'Select a source code location type'),
+        buttons: [autoDeployButton, ...makeButtons()],
+    })
+}
+
+export type CreateServiceRequest = Omit<AppRunner.CreateServiceRequest, 'SourceConfiguration'> & {
+    SourceConfiguration: CodeRepositorySource | ImageRepositorySource
+}
+
+export class CreateAppRunnerServiceForm extends WizardForm<CreateServiceRequest> {
```

Comment:
Just noticed we have Wizard and WizardForm. We could eliminate a concept by just merging them, instead of having Wizard contain a WizardForm.